src/coinbase/actions/sendUserOperation.ts

```diff
@@ -0,0 +1,114 @@
+import { UserOperationCalls } from "viem/_types/account-abstraction";
+import { SmartWallet } from "../wallets/types";
+import { NetworkIdentifier, UserOperationStatusEnum } from "../../client";
+import { encodeFunctionData, Hex } from "viem";
+import { Coinbase } from "../coinbase";
+import { wait } from "../utils/wait";
+
+export type SendUserOperationOptions<T extends readonly unknown[]> = {
+  calls: UserOperationCalls<T>;
+};
+
+export type SendUserOperationReturnType = {
+  id: string;
+  networkId: NetworkIdentifier;
+  smartWalletAddress: Hex;
+  status: UserOperationStatusEnum;
+  wait: () => Promise<SendUserOperationReturnType>;
+};
+
+export async function sendUserOperation<T extends readonly unknown[]>(
+  wallet: SmartWallet,
+  options: SendUserOperationOptions<T>,
+): Promise<SendUserOperationReturnType> {
+  const { networkId } = wallet;
+  if (!networkId) {
+    throw new Error("Network not set - call use({networkId}) first");
```

Comment:
don't love function calls in error messages incase we change the method, easy to miss and have bad errors